src/components/SafeRedirect.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { InjectAppServices } from '../services/pure-di';
+import RedirectToExternalUrl from './RedirectToExternalUrl';
+
+/**
+ * @param { Object } props
+ * @param { string } props.to
+ * @param { import('../services/pure-di').AppServices } props.dependencies
+ */
+function SafeRedirect({ to, dependencies }) {
+  const isCompleteUrl = (url) => {
+    return new RegExp(
+      /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/,
```

Comment:
What do you think about something more simple? like `/^\w+:/` if it matches, it is not a relative URL. 

Of course, it should be refined, but in my opinion, the focus should be on differentiating paths from other things..

What is more, maybe we could verify URLs against the white list, and if it does not match, simply assume that it is a legacy URL and concatenate with `dopplerLegacyUrl`.

Reply:
I thought that too, verify against a whitelist and that's it, but this could provide a bit more information in case of errors in urls as well, for example you added a wrongly form url in whitelist without noticing, perhaps I'm overthinking this...

Reply:
Nah, you are right, the problem is that you should deal with the errors hehe